### ISSUE TITLE
assert.strictEqual(false, ...) triggers unreachable code error

### ISSUE THREAD

**Issue by alexander-fenster** [microsoft/TypeScript#38699]
`assert.strictEqual(false, anyArgument)` triggers unreachable code error for any code below **after the recent @types/node update**. This **only** reproduces with `@types/node` 10.17.24, 12.12.41, 13.13.8 (published recently).

Note that it only reproduces with `false` as a value being checked. Replace it with anything else and it will compile.

**TypeScript Version:**  3.9.3, 3.8.x as well

**Search Terms:** `strictEqual` `error TS7027: Unreachable code detected`

**Code**

```ts
import * as assert from 'assert';

const b = false;
assert.strictEqual(false, b);
console.log('really unreachable?');
```

**Expected behavior:** should compile with `--allowUnreachableCode false`.

**Actual behavior:**

```
$ npx tsc --allowUnreachableCode false src/index.ts 
src/index.ts:5:1 - error TS7027: Unreachable code detected.

5 console.log('really unreachable?');
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.
```

**Playground Link:** N/A because of the specific version of `@types/node` required for a repro.

**Related Issues:** N/A

**Background:** Here is the commit in `@types/node` that apparently triggers this: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/e9103792668257b6cc86cd21d6c455ecb5fae1ea  
I reported it there as well in https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44944, but I have an impression the compiler behaves weirdly here.

**Comment by G-Rath** [comment]
Cross-posting from the DT issue for coverage:

This does look like it might be a bug in tsc, as while `true` & `false` are given a bit of extra special treatment, it still fails if both parameters are the `false` literal which doesn't seem right.

However, you can work around it in the meantime by casting false to boolean:

```
import * as assert from 'assert';

const b = false;

assert.strictEqual(false as boolean, b);
console.log('really unreachable?');
```

**Comment by ahejlsberg** [comment]
This is a bug in our code reachability analysis. It's a simple fix. Will put up a PR.

**Comment by ahejlsberg** [comment]
Ah, we already have a fix in #38710, but it never got merged. Will do now.
